From 4c47a36fb6e2e4349995376cee063bb37d4e68e3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 23 Jan 2020 15:57:05 +0100
Subject: [PATCH 2/2] Do not include lib/md5.c into src/shar.c
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

lib/md5.o is part of libgnu.a that is linked to shar. There is no
point in linking md5.o twice into shar executable. Moreover SuSE
reports that this triggers a linking error with GCC 10:

/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld:
../lib/libgnu.a(md5.o): in function `md5_stream':
[   30s] md5.c:(.text+0x15d): multiple definition of `md5_stream';
shar.o:shar.c:(.text+0x28): first defined here

<https://lists.gnu.org/archive/html/bug-gnu-utils/2020-01/msg00001.html>

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 src/shar.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/shar.c b/src/shar.c
index 11cbada..6d7ed1d 100644
--- a/src/shar.c
+++ b/src/shar.c
@@ -53,7 +53,6 @@ static const char cright_years_z[] =
 
 #include "inttostr.h"
 #include "liballoca.h"
-#include "md5.c"
 #include "md5.h"
 #include "quotearg.h"
 #include "xalloc.h"
-- 
2.21.1

